src/librustc_lint/types.rs

```diff
@@ -267,7 +267,19 @@ fn lint_int_literal<'a, 'tcx>(
         }
 
         cx.struct_span_lint(OVERFLOWING_LITERALS, e.span, |lint| {
-            lint.build(&format!("literal out of range for `{}`", t.name_str())).emit()
+            let mut err = lint.build(&format!("literal out of range for `{}`", t.name_str()));
+            err.note(&format!(
+                "the literal `{}` does not fit into an `{}` whose maximum is {}, minimum is {}",
```

Comment:
Did you consider `whose range is [{}, {}]`? That style is easier to parse for me.

Also, a small nit, I would not say "an u16". Is there a way to phrase the message without an indefinite article?